File: JavaScript/JS-OOP/JS-Classes/inheritance.py
class Shape:
    def __init__(self, color):
        self.color = color

    def move(self):
        print('move shape')

class Circle(Shape):
    def __init__(self, color, radius):
        super().__init__(color)
        self.radius = radius

    def draw(self):
        print('draw')

    # Overriding parent method
    def move(self):
        super().move()
        print('moving circle')

# Example usage:
c = Circle('black', 7)
c.move()  # Output: move shape
          #         moving circle
c.draw()  # Output: draw

File: JavaScript/JS-OOP/JS-Classes/inheritance.js
class Shape {
  constructor(color) {
    this.color = color;
  }
  move() {
    console.log("move shape");
  }
}

class Circle extends Shape {
  constructor(color, radius) {
    super(color);
    this.radius = radius;
  }
  draw() {
    console.log("draw");
  }
  //overriding parent method
  move() {
    super.move();
    console.log("moving circle");
  }
}

const c = new Circle("black", 7);
